clang/lib/DPCT/DPCT.cpp

```diff
@@ -762,9 +780,13 @@ int runDPCT(int argc, const char **argv) {
   ValidateInputDirectory(Tool, InRoot);
 
   IsUsingDefaultOutRoot = OutRoot.empty();
-  if (!makeOutRootCanonicalOrSetDefaults(OutRoot)) {
-    ShowStatus(MigrationErrorInvalidInRootOrOutRoot);
-    dpctExit(MigrationErrorInvalidInRootOrOutRoot, false);
+  if (DpctGlobalInfo::isQueryAPIMapping()) {
+    OutRoot = InRoot + "/dpct_output";
```

Comment:
Maybe we can use llvm/std filesystem `path` to do the concatenation.

Reply:
`llvm::sys::path::append` need the parameter's type: `SmallVectorImpl<char>`.

Reply:
remove this statment.